ext/storage_sources/s3_store/s3_connection.cpp

```diff
@@ -82,7 +82,7 @@ S3Connection::ListObjects(const std::string &prefix, std::vector<std::string> &o
             return (1);
         result = outcomes.GetResult();
         for (const auto &object : result.GetContents())
-            objects.push_back(object.GetKey());
+            objects.push_back(object.GetKey().substr(_objectPrefix.length());
```

Comment:
We are doing this at line 71, somehow we missed it here. The issue will only show up if there is a large number of objects to list.